contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+                _collateralAsset
+            ),
+            "OTCWrapper: insufficient collateral"
+        );
+
+        // user inflow
+        _depositWithPermit(
+            _userSignature.acct,
+            USDC,
+            _premium,
+            _userSignature.deadline,
+            _userSignature.v,
+            _userSignature.r,
+            _userSignature.s
+        );
+
+        // market maker inflow
+        _depositWithPermit(
+            _mmSignature.acct,
+            _collateralAsset,
+            _collateralAmount,
+            _mmSignature.deadline,
+            _mmSignature.v,
+            _mmSignature.r,
+            _mmSignature.s
+        );
+
+        // transfer fee to beneficiary address
+        IERC20(USDC).safeTransfer(beneficiary, orderFee);
+
+        // transfer premium to market maker only if collateral is not USDC
+        if (_collateralAsset != USDC) {
+            IERC20(USDC).safeTransfer(msg.sender, _premium.sub(orderFee));
+        }
+
+        // open vault
+        uint256 vaultID = (controller.getAccountVaultCounter(address(this))).add(1);
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](3);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.OpenVault,
+            address(this), // owner
+            address(this), // receiver
+            address(0), // asset, otoken
+            vaultID, // vaultId
+            0, // amount
+            0, // index
+            abi.encode(1) // vault type
+        );
+
+        // Approve margin pool to deposit collateral
+        IERC20(_collateralAsset).safeApproveNonCompliant(addressbook.getMarginPool(), totalInitialMargin);
+
+        // deposit collateral
+        actions[1] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.DepositCollateral,
+            address(this), // owner
+            address(this), // address to transfer from
+            _collateralAsset, // deposited asset
+            vaultID, // vaultId
+            totalInitialMargin, // amount
+            0, //index
+            "" //data
+        );
+
+        // create otoken, get an address and calculate mint amount
+        address otoken = IOtokenFactoryWrapperInterface(addressbook.getOtokenFactory()).createOtoken(
+            order.underlying,
+            USDC,
+            _collateralAsset,
+            order.strikePrice,
+            order.expiry,
+            order.isPut
+        );
+
+        // scales by 1e8 for division with oracle price
+        // scales by 1e2 to increase from 6 decimals (notional) to 8 decimals (otoken)
+        uint256 mintAmount = order.notional.mul(1e10).div(oracle.getPrice(order.underlying));
+
+        actions[2] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.MintShortOption,
+            address(this), // owner
+            address(this), // address to transfer to
```

Comment:
we should transfer oTokens directly to order.buyer

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/4874b61b9634b8d39db6741744a206cb673e6dc9